sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -174,23 +176,24 @@ def enabled(self, value):
         # type: (bool) -> bool
         if self._value is None:
             self._value = {"enabled": value}
-        elif isinstance(self._value, dict):
+        elif not isinstance(self._value, dict):
+            raise AttributeError("value parameter is expected to be a dictionary")
+        else:
             self._value["enabled"] = self._enabled
         self._enabled = value
 
     @property
     def value(self):
         # type: () -> Dict[str, Any]
-        self._enabled = self._value.get("enabled", self._enabled)
+        self._enabled = self._value.get("enabled") or self._enabled
         return self._value
 
     @value.setter
     def value(self, new_value):
         # type: (Dict[str, Any]) -> None
-        try:
-            self._enabled = new_value.get("enabled", self._enabled)
-        except AttributeError:
-            pass
+        if not new_value:
```

Comment:
Are None & empty string valid for new_value?

Reply:
No they are not.

Reply:
Why? I think we can set value to None or empty string for basic config settings?

Reply:
For `ConfigurationSetting` it is valid, but for `SecretReference` and `FeatureFlag` the `._value` has to be a dictionary.

Reply:
So it is a valid scenario that users treat it as basic config setting and set the value to empty string, right? We cannot silently change the value to {}.

Reply:
Yes, I am adding a test for it now.